### PR TITLE
[stdlib][Semantics annotation] Add ".empty" suffix to the semantics attribute of  array.init()  and add semantics annotation to the compiler intrinsic: Array._allocateUninitializedArray

### PR DESCRIPTION
**Pull request by ravikandhadai**
These semantics attribute allow the constant interpreter to distinguish between an empty array initializer and the compiler intrinsic: _allocateUninitializerArray

### CODE REVIEW

**Comment by ravikandhadai** [comment]
@marcrasi Based on @eeckstein suggestion, I have modified the semantics attribute of array.init() and added a semantics attribute for _allocateUninitializerArray. Once this is merged, the other PR can be rebased on top it.

**Comment by ravikandhadai** [comment]
@swift-ci Please test

**Comment by eeckstein** [comment]
@swift-ci benchmark

**Review by eeckstein — APPROVED** [review]
LGTM.
I started benchmarks to make sure there are no regressions. Changing semantics can have an effect on the optimizer, e.g. inlining

**Comment by ravikandhadai** [comment]
It looks like noise, as they don't seem to make use of arrays. Let me try again.

**Comment by ravikandhadai** [comment]
@swift-ci benchmark

**Comment by ravikandhadai** [comment]
@swift-ci Please test OS X Platform

**Comment by ravikandhadai** [comment]
@swift-ci Please test macOS Platform

**Comment by ravikandhadai** [comment]
@swift-ci Please test and merge

**Comment by ravikandhadai** [comment]
@swift-ci Please test and merge
